packages/netlify-cms-core/src/backend.js

```diff
@@ -449,7 +449,7 @@ class Backend {
     }
 
     const commitMessage = commitMessageFormatter('delete', config, { collection, slug, path });
-    return this.implementation.deleteFile(path, commitMessage);
+    return this.implementation.deleteFile(path, commitMessage, { collection, slug });
```

Comment:
I need a way to get the collection name for deletion. Every entry point in the backend API passes the collection name, which allows the unique identification of a collection rather than relying on the path to have the correct information.

The slug is a more reliable source rather than parsing it out of the path also.

This was the best way to have an API fix like `persistEntry`

```
persistEntry(entry, mediaFiles = [], options) {
```

```
deleteFile(path, commitMessage, options) {
```

This will allow for a non breaking change and also allow an API to call the `deleteFile`

Reply:
One more item to point out. `deleteUnpublishedEntry`, `publishUnpublishedEntry`, updateUnpublishedEntryStatus also passes these values as arguments. 
```
  deleteUnpublishedEntry(collectionName, slug) {
```
```
 publishUnpublishedEntry(collectionName, slug) {
```
```
updateUnpublishedEntryStatus(collectionName, slug, newStatus) { // newStatus should have been options.newStatus
```
During editorial workflow, we use these 2 values as the unique identifiers for an entry. It would be nice for the backend API to be more consistent, but I am not sure we want to do that rewrite at this time.

Reply:
@erquhart do you feel there is a better solution? This was about the clearest I could come up with without having to go back and search through a collection of entries trying to find a path.

Reply:
Sorry, just now circling back - this makes sense, agreed on all counts 👍